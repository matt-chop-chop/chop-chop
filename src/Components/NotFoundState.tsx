import { Button, Flex, Link, Text } from "@chakra-ui/react";
import { InfoOutlineIcon } from "@chakra-ui/icons";
import NextLink from "next/link";

const NotFoundState = () => {
  return (
    <Flex
      alignItems="center"
      background="var(--chakra-colors-light-background)"
      flexDirection="column"
      justifyContent="center"
      minHeight="250px"
      mt="100px"
      p={4}
      width="100%"
    >
      <Text textAlign="center" variant="body-bold">
        <InfoOutlineIcon mr={2} />
        This page could not be found. The URL entered did not match any pages on
        this site.
      </Text>
      <Text mt={2} variant="body">
        Adjust the URL entered or return to the homepage.
      </Text>
      <Link
        as={NextLink}
        href="/"
        mt={2}
        transition="transform 250ms ease-out"
        _hover={{
          transition: "transform 250ms ease-in",
          transform: "scale(1.1)",
        }}
      >
        <Button variant="primary">Home</Button>
      </Link>
    </Flex>
  );
};

export default NotFoundState;
