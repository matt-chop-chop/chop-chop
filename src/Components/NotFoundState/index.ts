export { default as NotFoundState } from "./NotFoundState";
