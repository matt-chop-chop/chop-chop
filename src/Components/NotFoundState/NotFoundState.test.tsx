import { shallow } from "enzyme";
import React from "react";
import NotFoundState from "./NotFoundState";

describe("NotFoundState", () => {
  it("is exported from NotFoundState", () => {
    NotFoundState;
  });

  it("renders as expected", () => {
    const wrapper = shallow(<NotFoundState />);

    expect(wrapper).toMatchSnapshot();
  });
});
