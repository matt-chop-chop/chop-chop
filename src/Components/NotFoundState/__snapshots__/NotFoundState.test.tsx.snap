// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFoundState renders as expected 1`] = `
<Flex
  alignItems="center"
  background="var(--chakra-colors-light-background)"
  flexDirection="column"
  justifyContent="center"
  minHeight="250px"
  mt="100px"
  p={4}
  width="100%"
>
  <Text
    textAlign="center"
    variant="body-bold"
  >
    <InfoOutlineIcon
      mr={2}
    />
    This page could not be found. The URL entered did not match any pages on this site.
  </Text>
  <Text
    mt={2}
    variant="body"
  >
    Adjust the URL entered or return to the homepage.
  </Text>
  <Link
    _hover={
      {
        "transform": "scale(1.1)",
        "transition": "transform 250ms ease-in",
      }
    }
    as={
      {
        "$$typeof": Symbol(react.forward_ref),
        "default": [Circular],
        "render": [Function],
      }
    }
    href="/"
    mt={2}
    transition="transform 250ms ease-out"
  >
    <Button
      variant="primary"
    >
      Home
    </Button>
  </Link>
</Flex>
`;
